rfq/manager.go

```diff
@@ -1132,97 +1149,67 @@ const (
-// public key.
+// RfqChannel returns the channel to use for RFQ operations. It returns a map of
+// peers and their eligible channels. If a peerPubKey is specified then the map
+// will only contain one entry for that peer.
 func (m *Manager) RfqChannel(ctx context.Context,
-	chanLister ChanLister,
-	specifier asset.Specifier, peerPubKey *route.Vertex,
-	intention chanIntention) (*ChannelWithSpecifier, error) {
+	chanLister ChanLister, specifier asset.Specifier,
+	peerPubKey *route.Vertex, intention ChanIntention) (map[route.Vertex][]ChannelWithSpecifier,
+	error) {
 
 	activeChannels, err := chanLister(ctx, true, false)
 	if err != nil {
 		return nil, err
 	}
 
-	balances, err := m.ComputeChannelAssetBalance(
+	balancesMap, err := m.ComputeChannelAssetBalance(
 		ctx, activeChannels, specifier,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("error computing available asset "+
 			"channel balance: %w", err)
 	}
 
-	if len(balances) == 0 {
+	if len(balancesMap) == 0 {
 		return nil, fmt.Errorf("no asset channel balance found for %s",
 			&specifier)
 	}
 
-	// If a peer public key was specified, we always want to use that to
-	// filter the asset channels.
-	if peerPubKey != nil {
-		balances = fn.Filter(
-			balances, func(c ChannelWithSpecifier) bool {
-				return c.ChannelInfo.PubKeyBytes == *peerPubKey
-			},
-		)
-	}
-
-	switch {
-	// If there are multiple asset channels for the same specifier, we need
-	// to ask the user to specify the peer public key. Otherwise, we don't
-	// know who to ask for a quote.
-	case len(balances) > 1 && peerPubKey == nil:
-		return nil, fmt.Errorf("multiple asset channels found for "+
-			"%s, please specify the peer pubkey", &specifier)
-
-	// We don't have any channels with that asset ID and peer.
-	case len(balances) == 0:
-		return nil, fmt.Errorf("no asset channel found for %s",
-			&specifier)
-	}
-
-	// If the user specified a peer public key, and we still have multiple
-	// channels, it means we have multiple channels with the same asset and
-	// the same peer, as we ruled out the rest of the cases above.
-
-	// Initialize best balance to first channel of the list.
-	bestBalance := balances[0]
-
 	switch intention {
-	case ReceiveIntention:
-		// If the intention is to receive, return the channel
-		// with the best remote balance.
-		fn.ForEach(balances, func(b ChannelWithSpecifier) {
-			if b.AssetInfo.RemoteBalance >
-				bestBalance.AssetInfo.RemoteBalance {
-
-				bestBalance = b
-			}
-		})
-
 	case SendIntention:
-		// If the intention is to send, return the channel with
-		// the best local balance.
-		fn.ForEach(balances, func(b ChannelWithSpecifier) {
-			if b.AssetInfo.LocalBalance >
-				bestBalance.AssetInfo.LocalBalance {
-
-				bestBalance = b
-			}
-		})
-
+		// When sending we care about the volume of our local balances,
+		// so we sort by local balances in descending order.
+		for _, v := range balancesMap {
+			sort.Slice(v, func(i, j int) bool {
```

Comment:
Hmm, will this actually mutate the underlying value here (the slice) captured by the loop? 

Would favor making the re-assignment explicit here, which may guard against future foot guns.